src/main/java/kitchenpos/application/TableGroupService.java

```diff
@@ -1,88 +1,68 @@
+    private final TableGroupRepository tableGroupRepository;
+
+    public TableGroupService(final OrderService orderService,
+        final TableService tableService,
+        final TableGroupRepository tableGroupRepository) {
+        this.orderService = orderService;
+        this.tableService = tableService;
+        this.tableGroupRepository = tableGroupRepository;
     }
 
     @Transactional
-    public TableGroup create(final TableGroup tableGroup) {
-        final List<OrderTable> orderTables = tableGroup.getOrderTables();
-
-        if (CollectionUtils.isEmpty(orderTables) || orderTables.size() < 2) {
-            throw new IllegalArgumentException();
-        }
-
-        final List<Long> orderTableIds = orderTables.stream()
-                .map(OrderTable::getId)
-                .collect(Collectors.toList());
-
-        final List<OrderTable> savedOrderTables = orderTableDao.findAllByIdIn(orderTableIds);
-
-        if (orderTables.size() != savedOrderTables.size()) {
-            throw new IllegalArgumentException();
-        }
-
-        for (final OrderTable savedOrderTable : savedOrderTables) {
-            if (!savedOrderTable.isEmpty() || Objects.nonNull(savedOrderTable.getTableGroupId())) {
-                throw new IllegalArgumentException();
-            }
-        }
+    public TableGroup create(final TableGroupRequest tableGroupRequest) {
+        tableGroupRequest.validateOrderTableSize();
 
-        tableGroup.setCreatedDate(LocalDateTime.now());
+        final List<OrderTable> orderTables = tableService.findAllByIdIn(tableGroupRequest.getOrderTableIds());
+        validateOrderTableEmpty(orderTables);
 
-        final TableGroup savedTableGroup = tableGroupDao.save(tableGroup);
-
-        final Long tableGroupId = savedTableGroup.getId();
-        for (final OrderTable savedOrderTable : savedOrderTables) {
-            savedOrderTable.setTableGroupId(tableGroupId);
-            savedOrderTable.setEmpty(false);
-            orderTableDao.save(savedOrderTable);
-        }
-        savedTableGroup.setOrderTables(savedOrderTables);
-
-        return savedTableGroup;
+        final TableGroup tableGroup = new TableGroup(orderTables);
+        tableGroup.validateEqualOrderTableSize(orderTables.size());
+        tableGroup.updateOrderTables(orderTables);
+        return tableGroup;
     }
 
+
     @Transactional
     public void ungroup(final Long tableGroupId) {
-        final List<OrderTable> orderTables = orderTableDao.findAllByTableGroupId(tableGroupId);
+        final TableGroup tableGroup = findTableGroupById(tableGroupId);
+        final List<OrderTable> orderTables = tableService.findAllByTableGroup(tableGroup);
 
-        final List<Long> orderTableIds = orderTables.stream()
-                .map(OrderTable::getId)
-                .collect(Collectors.toList());
+        orderService.validateOrderStatusNotIn(orderTables, Arrays.asList(OrderStatus.COOKING, OrderStatus.MEAL));
```

Comment:
`List<OrderTable>` 조회하기 보다는 연관관계를 활용해볼 수 있겠네요.

```suggestion
        orderService.validateOrderStatusNotIn(tableGroup.getOrderTables(), Arrays.asList(OrderStatus.COOKING, OrderStatus.MEAL));
```